lightning/src/ln/channelmanager.rs

```diff
@@ -17664,6 +17665,21 @@ where
 					let mut peer_state_lock = peer_state_mtx.lock().unwrap();
 					let peer_state = &mut *peer_state_lock;
 					is_channel_closed = !peer_state.channel_by_id.contains_key(channel_id);
+					if let Some(chan) = peer_state.channel_by_id.get(channel_id) {
+						if let Some(funded_chan) = chan.as_funded() {
+							let inbound_committed_update_adds =
+								funded_chan.get_inbound_committed_update_adds();
+							if !inbound_committed_update_adds.is_empty() {
+								// Reconstruct `ChannelManager::decode_update_add_htlcs` from the serialized
+								// `Channel`, as part of removing the requirement to regularly persist the
+								// `ChannelManager`.
+								decode_update_add_htlcs.insert(
```

Comment:
Can't this be done in the next loop over the monitors where the update_add_htlcs are deduped? This loop seems to be about payments.

Reply:
Hmm, IMO the first loop is about repopulating HTLC maps from monitors, and the second loop is about pruning maps from monitors, so this approach seems to fit. Also getting test failures with that approach, I think we can hit weird behavior if we try to rebuild and prune in the same loop, which is why there are two loops to begin with.

Reply:
I thought the first loop is about repopulating (outbound) payments. We probably mean the same thing.

Reply:
Oh interesting, didn't think of that. Do you want to add that explanation as a code comment, so that the knowledge is recorded?

Reply:
Addressed in #4289